### PR TITLE
add surv.dcalib

### PR DESCRIPTION
**Pull request by RaphaelS1**
Need to check the maths on this measure before merging, looks reasonable but fairly new so want to double check.

https://www.jmlr.org/papers/volume21/18-772/18-772.pdf

### CODE REVIEW

**Comment by RaphaelS1** [comment]
@adibender would you mind also reviewing this? It's the same maths behind the plot in mlr3viz

File: R/MeasureSurvDCalibration.R [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    .B = 10L,
+    .chisq = FALSE,
+    .score = function(prediction, ...) {
+
+      # initialize buckets
+      bj = numeric(self$B)
+      # predict individual probability of death at observed event time
+      si = as.numeric(prediction$distr$survival(data = matrix(prediction$truth[, 1L], nrow = 1L)))
+      # remove negative values
+      si = sapply(si, function(.x) max(.x, 1e-5))
```
Comment (mllg) [inline R/MeasureSurvDCalibration.R:88]: `map_dbl()`

File: R/MeasureSurvDCalibration.R [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    initialize = function(B = 10L, chisq = FALSE) {
+      super$initialize(
+        id = "surv.dcalib",
+        range = c(0, Inf),
+        minimize = TRUE,
+        predict_type = "distr",
+        man = "mlr3proba::mlr_measures_surv.dcalib",
+      )
+
+      private$.B = assertIntegerish(B)
```
Comment (mllg) [inline R/MeasureSurvDCalibration.R:48]: `assert_integerish()` if you want to stick to the mlr3 style. But I don't really care.

**Review by mllg — APPROVED** [review]

**Comment by mllg** [comment]
LGTM.

**Pull request merged**